lib/ht_traject/print_holdings_api.rb

```diff
@@ -1,15 +1,18 @@
 require_relative "../services"
+require "httpx/adapters/faraday"
 
 module HathiTrust
   class PrintHoldingsAPI
 
     def self.connection 
-      @conn ||= Faraday.new( url: ENV["HOLDINGS_API_URL"] ) do |builder|
+      Thread.current[:holdings_api_conn] ||= Faraday.new( url: ENV["HOLDINGS_API_URL"] ) do |builder|
```

Comment:
This is required to make sure that each thread has its own connection pool and the threads aren't potentially getting clogged up trying to re-use the same connection.